### PR TITLE
TransferUtilityDownloadDirectoryRequest is missing SSE-C properties

### PR DESCRIPTION
**Pull request by lrocher**
TransferUtilityDownloadDirectoryRequest is missing the three SSE-C properties:
TransferUtilityUploadDirectoryRequest  already manage this feature.

## Description

Add the three ServerSideEncryptionCustomer properties (Method, Key, KeyMD5) to the TransferUtilityUploadDirectoryRequest. And pass them in the DownloadDirectoryCommand to the TransferUtilityUploadDownload..

## Motivation and Context

fixes https://github.com/aws/aws-sdk-net/issues/3147

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [X] I have updated the documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues/3147
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: sdk/src/Services/S3/Custom/Transfer/Internal/DownloadDirectoryCommand.cs [hunk 1]
```diff
@@ -98,7 +98,10 @@ private TransferUtilityDownloadRequest ConstructTransferUtilityDownloadRequest(S
             downloadRequest.Key = s3Object.Key;
             var file = s3Object.Key.Substring(prefixLength).Replace('/', Path.DirectorySeparatorChar);
             downloadRequest.FilePath = Path.Combine(this._request.LocalDirectory, file);
-                        
+            downloadRequest.ServerSideEncryptionCustomerMethod = this._request.ServerSideEncryptionCustomerMethod;
```
Comment (dscpinheiro) [inline sdk/src/Services/S3/Custom/Transfer/Internal/DownloadDirectoryCommand.cs:101]: Could you add a new integration test here? https://github.com/aws/aws-sdk-net/blob/main/sdk/test/Services/S3/IntegrationTests/TransferUtilityTests.cs

Here's one example for uploading objects (to a bucket that requires `SSE-C`): https://github.com/aws/aws-sdk-net/blob/3.7.729.0/sdk/test/Services/S3/IntegrationTests/TransferUtilityTests.cs#L438-L464

**Comment by lrocher** [comment]
I try to add a Test for TransferUtility UploadDirectory and DownloadDirectory with SSE-C.

File: sdk/test/Services/S3/IntegrationTests/TransferUtilityTests.cs [hunk 2]
```diff
@@ -230,6 +230,55 @@ public void SimpleUploadWithSSE_C_LargeFile()
...
+                LocalDirectory = downloadPath,
+
+                ServerSideEncryptionCustomerMethod = ServerSideEncryptionCustomerMethod.AES256,
+                ServerSideEncryptionCustomerProvidedKey = base64Key
+            };
+
+            transferUtility.DownloadDirectory(requestDownload);
+
+            // Compare directories
+            UtilityMethods.CompareFiles(directoryTestPath, downloadPath);
```
Comment (dscpinheiro) [inline sdk/test/Services/S3/IntegrationTests/TransferUtilityTests.cs:279]: I ran this test locally and it failed, it looks like our `UtilityMethods.CompareFiles` method doesn't work when you pass a directory path (instead of a file path).

Changing that method is out of scope for this PR, so I think we could replace this section with something like this:
```
// Compare each file in both directories
var sourceFiles = Directory.EnumerateFiles(directoryTestPath, "*", SearchOption.AllDirectories).ToList();
var downloadedFiles = Directory.EnumerateFiles(downloadPath, "*", SearchOption.AllDirectories).ToList();

Assert.AreEqual(sourceFiles.Count, downloadedFiles.Count);
for (var i = 0; i < sourceFiles.Count(); i++)
{
    UtilityMethods.CompareFiles(sourceFiles[i], downloadedFiles[i]);
}
```

Comment (lrocher) [reply sdk/test/Services/S3/IntegrationTests/TransferUtilityTests.cs:279]: Oh sorry, i miss understanding usage of UtilityMethods.CompareFiles.
Solution is soo big, i'm not able to open it or run it.
I update PR soon

Comment (dscpinheiro) [reply sdk/test/Services/S3/IntegrationTests/TransferUtilityTests.cs:279]: No problem, the `AWSSDK.Net*.sln` solutions include all services so they're slower to load. 

We do create separate solutions for each service, the S3 one is located at `.\sdk\src\Services\S3\S3.sln` (that's how I ran this specific test).

**Review by dscpinheiro — APPROVED** [review]

**Review by CamileDahdah — APPROVED** [review]

**Comment by dscpinheiro** [comment]
Thanks again for the contribution, this will be included in the next SDK release.
